### ISSUE TITLE
Add PDF download button (not link) guidance and add to the pattern library

### ISSUE THREAD

**Issue by allison0034** [department-of-veterans-affairs/vets-design-system-documentation#670]
# Feature Request

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## This update is for:

- [x] Content style guide
- [x] Component
- [x] Pattern
- [ ] Utility
- [ ] Other

## What is the name?

The PDF download pattern. Not all PDF downloads should be a link, in instances where it doesn't just link to a static resource but instead, it fires off some JS that uses your information to generate your own personal document it should be a button.

## What is the nature of this update?

- [x] How to build this component/pattern
- [x] When to use this component/pattern 
- [x] When to use something else
- [ ] Usability guidance
- [x] Accessibility
- [ ] Implementation
- [ ] Research insights
- [ ] Package information

## What problem does this solve?

It solves an accessibility issue when PDFs use JS to create a document

## Additional Context

Per a discussion between Brian D, Josh Kim, and Martha, about [1095B](https://github.com/department-of-veterans-affairs/va.gov-team/issues/38351), it is recommended that we add this documentation...

- Recommend a button as the most materially honest option
- Recommend that they use some different text with the button to better convey what's happening. Maybe something like "Generate 1095-B and begin PDF download"?
- Bring this up as something to formalize in the design system -- links to download static PDFs, buttons to generate PDFs, standardized language.

We should also add the PDF download pattern to the pattern library, and call out the differences on when to use a link vs button as suggested above. Currently the [documentation](https://design.va.gov/content-style-guide/links) is in the Content section. I would rec adding a link to the new PDF pattern on that page as well.

**Comment by GnatalieH** [comment]
I just created guidance on PDF download links as part of the Links (Component) documentation I'm working on. I'm a little surprised we want to use a button for PDF downloads. We have so many PDF download links on va.gov, I'm hesitant to recommend a different pattern now. And, using lengthy text on a button isn't really feasible.

**Comment by GnatalieH** [comment]
I just documented a pattern for PDF download links in the Links (component) documentation I'm working on. I am not sure where this additional PDF as a button documentation would live, but it worries me a little to have divergent patterns for downloading a PDF. I realize this download pattern is different because the PDF is generated using JS, but does that difference matter to users? Playing devil's advocate here, perhaps I don't understand the 1095B PDF experience.

**Comment by allison0034** [comment]
@GnatalieH I agree with your concerns, however this is directly related to an accessibility issue per Josh, Martha and Brian. I think this is where we would need to be clear in our documentation. Most our PDFs will be a text link, however if the PDF is using js then in order to comply with accessibility it would need to be a button and the text would need to be descriptive for visual impairment, at least that is my understanding.

**Comment by GnatalieH** [comment]
@allison0034 thanks for explaining that!

**Comment by humancompanion-usds** [comment]
I had thought I had addressed this when I worked on Link but looking at it again both Link and Button could use an update. A couple of things still to do here:

* [ ] Under https://design.va.gov/components/link/#pdf-download-link we should clarify that the link treatment is only if linking directly to a PDF, not as a trigger for a process that generates a PDF. Also, link to the new section I added on [Links vs. Buttons](https://design.va.gov/components/link/action#links-vs-buttons) in Link - Action.
* [ ] While I did incorporate the [Content considerations from the Content Style Guide](https://design.va.gov/components/link/#content-considerations) on the new Link page, the last bullet conflicts with the PDF variation and should just link to it.
* [ ] Take this line from the Button page: "Buttons vs text links can be confusing. A good rule is if the action changes the url, it should not be a button." and add it to [Links vs. Buttons](https://design.va.gov/components/link/action#links-vs-buttons) but then make that section a partial shared between Link and Button components.
